src/components/search/__test__/LegalSearch.test.tsx

```
@@ -0,0 +1,168 @@
+        const thenMock = vi.fn().mockImplementation((callback) => {
+            callback({ data: mockDocuments, error: null });
+            return { data: mockDocuments, error: null };
+        });
+
+        const mockQueryBuilder = {
+            select: vi.fn().mockReturnThis(),
+            ilike: vi.fn().mockReturnThis(),
+            then: thenMock
+        };
+
+        (supabase.from as any).mockReturnValue(mockQueryBuilder);
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)